flocker/node/agents/test/test_blockdevice_manager.py

```diff
@@ -157,3 +167,102 @@ def test_formatted_bad_type(self):
+            filename = str(uuid4())
+            new_file = create.child(filename)
+            new_file.touch()
+            self.expectThat(view.child(filename).exists(), Equals(True),
+                            'Created file not visible through bind mount.')
+
+    def test_failing_bind_mount(self):
+        """
+        Attempts to bind mount to a mountpoint that does not exist fail with a
+        ``BindMountError``.
+        """
+        src_directory = self._get_directory_for_mount()
+        target_directory = self._get_directory_for_mount().child('nonexistent')
+        with ExpectedException(BindMountError, '.*nonexistent.*'):
+            self.manager_under_test.bind_mount(src_directory, target_directory)
+
+    def test_remount(self):
+        """
+        Mounts remounted read-only cannot be written to until they are
+        remounted read-write.
+        """
+        blockdevice = self._get_free_blockdevice()
+        mountpoint = self._get_directory_for_mount()
+        self.manager_under_test.make_filesystem(blockdevice, 'ext4')
+        self.manager_under_test.mount(blockdevice, mountpoint)
+
+        first_file = mountpoint.child(str(uuid4()))
+        second_file = mountpoint.child(str(uuid4()))
+        third_file = mountpoint.child(str(uuid4()))
+
+        first_file.touch()
+        self.expectThat(first_file.exists(), Equals(True))
+
+        self.manager_under_test.remount(mountpoint, Permissions.READ_ONLY)
+        self.expectThat(first_file.exists(), Equals(True))
+
+        with ExpectedException(OSError):
+            second_file.touch()
+        self.expectThat(second_file.exists(), Equals(False))
+
+        self.manager_under_test.remount(mountpoint, Permissions.READ_WRITE)
+        self.expectThat(first_file.exists(), Equals(True))
+        self.expectThat(second_file.exists(), Equals(False))
+        third_file.touch()
+        self.expectThat(third_file.exists(), Equals(True))
+
+    def test_remount_failure(self):
+        """
+        Remounts of a folder that is not mounted fail with ``RemountError``.
+        """
+        unmounted_directory = self._get_directory_for_mount()
+        with ExpectedException(RemountError):
+            self.manager_under_test.remount(unmounted_directory,
+                                            Permissions.READ_WRITE)
+
+    def test_make_tmpfs_mount(self):
+        """
+        make_tmpfs_mount should create a tmpfs mountpoint that can be written
+        to. Once the mount is unmounted all files should be gone.
+        """
+        mountpoint = self._get_directory_for_mount()
+        test_filename = unicode(uuid4())
+
+        # Twisted caches FilePath.exists, so if run path.exists() before and
+        # after you unmount a mountpoint, the FilePath will lie to you and
+        # claim that it still exists after the tmpfs mount was unmounted.
+        # Instead, create a new FilePath object every time to avoid bugs in the
+        # state of FilePath.
```

Comment:
I _think_ calling the `restat` method (see [docs](http://twistedmatrix.com/documents/current/api/twisted.python.filepath.FilePath.html#restat)) will do what you want here.